circuits/cpp/barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/eccvm/eccvm_circuit_builder.test.cpp

```diff
@@ -0,0 +1,273 @@
+    circuit.mul_accumulate(b, x);
+    circuit.mul_accumulate(c, x);
+
+    bool result = circuit.check_circuit();
+    EXPECT_EQ(result, true);
+}
+
+TYPED_TEST(ECCVMCircuitBuilderTests, Add)
+{
+    using Flavor = TypeParam;
+    using G1 = typename Flavor::CycleGroup;
+    proof_system::ECCVMCircuitBuilder<Flavor> circuit;
+
+    auto generators = G1::template derive_generators<1>();
+    typename G1::element a = generators[0];
+
+    circuit.add_accumulate(a);
+
+    bool result = circuit.check_circuit();
+    EXPECT_EQ(result, true);
+}
+
+TYPED_TEST(ECCVMCircuitBuilderTests, Mul)
+{
+    using Flavor = TypeParam;
+    using G1 = typename Flavor::CycleGroup;
+    using Fr = typename G1::Fr;
+    proof_system::ECCVMCircuitBuilder<Flavor> circuit;
+    auto generators = G1::template derive_generators<3>();
+    typename G1::element a = generators[0];
+    Fr x = Fr::random_element(&engine);
+
+    circuit.mul_accumulate(a, x);
+
+    bool result = circuit.check_circuit();
+    EXPECT_EQ(result, true);
+}
+
+TYPED_TEST(ECCVMCircuitBuilderTests, ShortMul)
+{
+    using Flavor = TypeParam;
+    using G1 = typename Flavor::CycleGroup;
+    using Fr = typename G1::Fr;
+    proof_system::ECCVMCircuitBuilder<Flavor> circuit;
+    auto generators = G1::template derive_generators<3>();
+
+    typename G1::element a = generators[0];
+    uint256_t small_x = 0;
+    // make sure scalar is less than 127 bits to fit in z1
+    small_x.data[0] = engine.get_random_uint64();
+    small_x.data[1] = engine.get_random_uint64() & 0xFFFFFFFFFFFFULL;
+    Fr x = small_x;
+
+    circuit.mul_accumulate(a, x);
+    circuit.eq(a * small_x);
+
+    bool result = circuit.check_circuit();
+    EXPECT_EQ(result, true);
+}
+
+TYPED_TEST(ECCVMCircuitBuilderTests, EqFails)
+{
+    using Flavor = TypeParam;
+    using G1 = typename Flavor::CycleGroup;
+    using Fr = typename G1::Fr;
+    proof_system::ECCVMCircuitBuilder<Flavor> circuit;
+
+    auto generators = G1::template derive_generators<3>();
+    typename G1::element a = generators[0];
+    Fr x = Fr::random_element(&engine);
+
+    circuit.mul_accumulate(a, x);
+    circuit.eq(a);
+    bool result = circuit.check_circuit();
+    EXPECT_EQ(result, false);
+}
+
+TYPED_TEST(ECCVMCircuitBuilderTests, EmptyRow)
+{
+    using Flavor = TypeParam;
+    proof_system::ECCVMCircuitBuilder<Flavor> circuit;
+
+    circuit.empty_row();
+
+    bool result = circuit.check_circuit();
+    EXPECT_EQ(result, true);
+}
+
+TYPED_TEST(ECCVMCircuitBuilderTests, EmptyRowBetweenOps)
+{
+    using Flavor = TypeParam;
+    using G1 = typename Flavor::CycleGroup;
+    using Fr = typename G1::Fr;
+    proof_system::ECCVMCircuitBuilder<Flavor> circuit;
+
+    auto generators = G1::template derive_generators<3>();
+    typename G1::element a = generators[0];
+    Fr x = Fr::random_element(&engine);
+
+    typename G1::element expected_1 = (a * x);
+
+    circuit.mul_accumulate(a, x);
+    circuit.empty_row();
+    circuit.eq(expected_1);
```

Comment:
Why is this the intended result?

Reply:
empty_row does nothing, so the value of the msm accumulator should still be [a] * x